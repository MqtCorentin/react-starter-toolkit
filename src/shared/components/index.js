export { default as ModalCommonBody } from './ModalCommonBody';
export { default as ModalCommonHeader } from './ModalCommonHeader';
export { default as ModalCommonFooter } from './ModalCommonFooter';
