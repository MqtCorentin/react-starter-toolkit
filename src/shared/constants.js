import { string, number, shape, oneOfType, arrayOf } from 'prop-types';

export const MSG_REQUIRED = 'Le champ est obligatoire';

export const MSG_FORMAT = 'Format invalide';

export const MSG_ERROR_FORMAT = 'Format de fichier incorrect';

export const rulesRequired = {
  required: {
    message: MSG_REQUIRED,
  },
};

export const rulePattern = {
  pattern: {
    regex: /^[0-9a-zA-Z\s-_]{1,50}$/,
    message: MSG_FORMAT,
  },
};

export const rulesErrorFormat = {
  required: {
    message: MSG_ERROR_FORMAT,
  },
};

export const STATUS_HTTP = {
  SUCCESS: 200,
  CREATED: 201,
  BAD_REQUEST: 400,
  UNAUTHORIZE: 401,
  FORBIDDEN: 403,
  NOTFOUND: 404,
  SERVER_ERROR: 500,
};

export const STATUS_API = {
  SUCCESS: STATUS_HTTP.SUCCESS,
  CREATED: STATUS_HTTP.CREATED,
  WARNING: 40,
  ERROR: 50,
};

export const STATUS_HTTP_MESSAGES = {
  [STATUS_HTTP.SUCCESS]: 'Succès: La requête a réussi',
  [STATUS_HTTP.CREATED]: 'Succès: Création réussie',
  [STATUS_HTTP.BAD_REQUEST]: 'Erreur: La syntaxe de la requête est erronée',
  [STATUS_HTTP.UNAUTHORIZE]: 'Erreur: Une authentification est nécessaire pour accéder à la ressource',
  [STATUS_HTTP.FORBIDDEN]: 'Erreur: Accès non autorisé',
  [STATUS_HTTP.NOTFOUND]: 'Erreur: Elément non trouvé',
  [STATUS_HTTP.SERVER_ERROR]: 'Erreur: Problème technique ! Contacter votre support',
};

export const TIMEOUT = 20000;
export const TIMEOUT_ERROR_MESSAGE = 'Erreur: la requête a mis trop de temps';

export const MODIFIER_CLASS = {
  success: 'success',
  disabled: 'disabled',
};

export const COMMENT = {
  NAME: 'comment',
  LABEL: 'Commentaire',
  SUBLABEL: 'Facultatif',
  MAXLENGTH: '300',
};

export const JSON_CONTENT_HEADER = {
  'Content-Type': 'application/json',
};

export const STORYBOOK = 'https://axaguildev.github.io/react-toolkit/latest/storybook/';
export const DESIGN_SYSTEM = 'https://axaguildev.github.io/design-system/';

export const ADMIN = 'Admin';
export const READER = 'Reader';
export const WRITER = 'Writer';
export const USER = 'User';

export const PROFILS = [ADMIN, READER, WRITER, USER];

export const DEFAULT_OPTION_LABEL = '- Sélectionner -';
export const DEFAULT_OPTION = {
  value: '',
  label: DEFAULT_OPTION_LABEL,
};

/**
 * Commons proptypes
 */

export const ContextPropTypes = shape({
  Provider: shape({ defaultValue: string, currentValue: string, changedBits: number }),
  Consumer: shape({ defaultValue: string, currentValue: string, changedBits: number }),
});

export const SelectPropTypes = arrayOf(
  shape({
    value: oneOfType([string, number]),
    label: string,
  }),
);

export const AnomalyPropTypes = shape({
  code: number,
  detail: string,
  label: string,
});

export const IdPropTypes = oneOfType([string, number]);
