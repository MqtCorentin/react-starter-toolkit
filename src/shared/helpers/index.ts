export { formatDate, isValidDate, setDate } from './formatDate';
export { default as setInitialState } from './setInitialState';
export { default as isEmptyOrNull } from './isEmptyOrNull';
export { default as mergeObj } from './mergeObj';
export { default as setResponseError } from './setResponseError';
export { default as setAnomalyEmptyItems } from './setAnomalyEmptyItems';
export { default as setConfirmClassModifier } from './setConfirmClassModifier';
export { default as setLoaderMode } from './setLoaderMode';
export { genericHandleChange, getErrorsList, computeInitialStateErrorMessage, getValuesList } from './validation.generic';
export { setTextTooltip, setDisplay } from './formatDataTable';
