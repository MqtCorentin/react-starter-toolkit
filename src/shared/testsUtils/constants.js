export const MOCK_API_URL = 'https://react-starter-api.vercel.app/api/';
