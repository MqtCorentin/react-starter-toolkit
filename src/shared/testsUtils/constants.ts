export const MOCK_API_URL = { vercel: 'https://react-starter-api.vercel.app/api/', github: 'https://raw.githubusercontent.com/' };
