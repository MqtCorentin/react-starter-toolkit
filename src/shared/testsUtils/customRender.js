import { render } from '@testing-library/react';
import { StaticRouter } from 'react-router-dom/server';
import FetchProvider from 'App/FetchProvider';
import UserProvider from 'App/UserProvider';
import { MOCK_API_URL } from '.';

const MockProviders =
  ({ role, ...testMock }) =>
  ({ children }) => {
    const useOidcAccessTokenFn = jest.fn().mockReturnValue({ accessToken: 'accessToken' });
    const useOidcUserFn = jest.fn().mockReturnValueOnce({ oidcUser: { member_of: [`CN=${role}`] } });
    return (
      <UserProvider useOidcUserFn={useOidcUserFn}>
        <FetchProvider
          apiUrl={MOCK_API_URL}
          fetchConfig={{
            headers: { testMock: JSON.stringify(testMock) },
          }}
          useOidcAccessTokenFn={useOidcAccessTokenFn}
        >
          <StaticRouter context={{}}>{children}</StaticRouter>
        </FetchProvider>
      </UserProvider>
    );
  };

const customRender = (ui, options, testMock) => render(ui, { wrapper: MockProviders(testMock), ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
