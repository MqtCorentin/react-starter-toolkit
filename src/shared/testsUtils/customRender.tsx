import { render, RenderOptions } from '@testing-library/react';
import { StaticRouter } from 'react-router-dom/server';
import FetchProvider from 'App/FetchProvider';
import UserProvider from 'App/UserProvider';
import { ReactElement, ReactNode } from 'react';
import { MOCK_API_URL } from '.';

type TMockProvider = {
  [x: string]: Record<string, unknown | number | string> | string;
};

const MockProviders =
  ({ role = '', ...testMock }: TMockProvider) =>
  ({ children }: { children: ReactNode }) => {
    const useOidcAccessTokenFn = jest.fn().mockReturnValue({ accessToken: 'accessToken' });
    const useOidcUserFn = jest.fn().mockReturnValueOnce({ oidcUser: { member_of: [`CN=${role}`], name: 'Samuel Gomez' } });
    return (
      <UserProvider useOidcUserFn={useOidcUserFn}>
        <FetchProvider
          apiUrl={MOCK_API_URL}
          fetchConfig={{
            headers: { testMock: JSON.stringify(testMock) },
          }}
          useOidcAccessTokenFn={useOidcAccessTokenFn}
        >
          <StaticRouter location="">{children}</StaticRouter>
        </FetchProvider>
      </UserProvider>
    );
  };

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>, testMock: TMockProvider = { role: '' }) =>
  render(ui, { wrapper: MockProviders(testMock), ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
