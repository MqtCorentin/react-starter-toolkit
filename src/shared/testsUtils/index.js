export { default as emptyFunction } from './emptyFunction';
export { default as WrapperStaticRouter } from './WrapperStaticRouter';
export { default as renderWithWrapperStaticRouter } from './renderWithWrapperStaticRouter';
