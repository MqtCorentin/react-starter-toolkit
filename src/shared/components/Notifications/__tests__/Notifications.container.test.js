import { render } from '@testing-library/react';
import { emptyFunction } from 'shared/testsUtils';
import NotificationsEnhanced from '../Notifications.container';

describe('NotificationsEnhanced', () => {
  it('Render <NotificationsEnhanced/>', () => {
    const defaultProps = {
      className: '',
      notifications: [],
      deleteNotification: emptyFunction,
    };
    const { asFragment } = render(<NotificationsEnhanced {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Render <NotificationsEnhanced/> with notifications', () => {
    const defaultProps = {
      className: '',
      notifications: [{ id: 'id', title: 'label' }],
      deleteNotification: emptyFunction,
    };
    const { asFragment } = render(<NotificationsEnhanced {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
