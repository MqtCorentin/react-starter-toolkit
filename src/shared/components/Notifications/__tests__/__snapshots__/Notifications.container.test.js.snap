// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationsEnhanced Render <NotificationsEnhanced/> 1`] = `
<DocumentFragment>
  <aside
    class=""
  />
</DocumentFragment>
`;

exports[`NotificationsEnhanced Render <NotificationsEnhanced/> with notifications 1`] = `
<DocumentFragment>
  <aside
    class=""
  >
    <div
      class="af-alert af-alert--notification af-alert--error"
      role="alert"
    >
      <div
        class="af-alert__title"
      >
        <div
          class="af-alert__title-icon"
        >
          <i
            class="glyphicon glyphicon-minus-sign"
          />
        </div>
        <div
          class="af-alert__title-text"
        >
          label
        </div>
        <button
          class="af-alert__title-icon-close"
          type="button"
        >
          <i
            class="glyphicon glyphicon-close"
          />
        </button>
      </div>
    </div>
  </aside>
</DocumentFragment>
`;
